.pre-commit-config.yaml

```diff
@@ -28,6 +28,16 @@ repos:
   -   id: check-merge-conflict
   -   id: check-executables-have-shebangs
   -   id: check-shebang-scripts-are-executable
+- repo: https://github.com/ssciwr/clang-format-hook.git
+  rev: v14.0.0
+  hooks:
+  -   id: clang-format
+      types_or: [c++, c]
+- repo: https://github.com/cpplint/cpplint.git
+  rev: 1.6.1
+  hooks:
+  -   id: cpplint
+      args: ['--quiet', '--filter=-whitespace/comments', '--linelength=120']
```

Comment:
How do you feel about `clang-tidy` as a unit test instead of `cpplint` as a `pre-commit` check?

Reply:
It was not as easy as I was expecting it to be. Naturally, ROS 2 generated code won't pass `clang-tidy`. `ament_cmake_clang_tidy` wraps `clang-tidy` in enough abstraction layers to make it impossible to narrow down the set of source files it diagnoses.

So, I hooked it up manually in https://github.com/bdaiinstitute/ros_utilities/pull/118/commits/7513af2fa4a8dc7a38ec97060f962c2455c1851d. I also kept `cpplint`. These are not mutually exclusive.

Reply:
They are not, but I've found that cpplint causes a lot more comments to ignore linting than one would like. I don't think its a problem to have it on though.